src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs

```diff
@@ -505,8 +507,8 @@ public virtual void GenerateAnonymousFunctionConversionError(BindingDiagnosticBa
         // Returns the inferred return type, or null if none can be inferred.
         public BoundLambda Bind(NamedTypeSymbol delegateType)
         {
-            BoundLambda result;
-            if (!_bindingCache.TryGetValue(delegateType, out result))
+            BoundLambda? result;
+            if (!_bindingCache!.TryGetValue(delegateType, out result))
```

Comment:
>!. [](start = 30, length = 2)

`Debug.Assert`?

Reply:
Since we're directly dereferencing the value, an assert would only add clutter.